python/cris_import.py

```diff
@@ -0,0 +1,257 @@
+        row["crash_date"] = crash_date_iso
+        row["crash_time"] = crash_date_iso
+
+
+def chunks(lst, n):
+    """Yield successive n-sized chunks from lst."""
+    for i in range(0, len(lst), n):
+        yield lst[i : i + n]
+
+
+def main():
+    files_todo = get_files_todo()
+    extract_ids = list(set([f["extract_id"] for f in files_todo]))
+    # assumes extract ids are sortable oldest > newest by filename. todo: is that right?
+    extract_ids.sort()
+    for extract_id in extract_ids:
+        # get schema years that match this extract ID
+        schema_years = list(
+            set([f["schema_year"] for f in files_todo if f["extract_id"] == extract_id])
+        )
+        schema_years.sort()
+        for schema_year in schema_years:
+            for table_name in ["crashes", "units", "persons"]:
+                file = next(
+                    (
+                        f
+                        for f in files_todo
+                        if f["extract_id"] == extract_id
+                        and f["schema_year"] == schema_year
+                        and table_name.startswith(f["table_name"])
+                    ),
+                    None,
+                )
+
+                if not file:
+                    raise (
+                        f"No {table_name} file found in extract. This should never happen!"
+                    )
+
+                print(f"processing {table_name}")
+
+                records = handle_empty_strings(
+                    remove_unsupported_columns(
+                        lower_case_keys(load_csv(file["path"])), table_name
+                    )
+                )
+
+                if table_name == "crashes":
+                    handle_crash_date_time(records)
+
+                # annoying redundant branch to combine primary persons and persons
+                if table_name == "persons":
+
+                    p_person_file = next(
+                        (
+                            f
+                            for f in files_todo
+                            if f["extract_id"] == extract_id
+                            and f["schema_year"] == schema_year
+                            and f["table_name"].startswith("primaryperson")
+                        ),
+                        None,
+                    )
+
+                    for p in records:
+                        p["is_primary"] = False
+
+                    pp_records = handle_empty_strings(
+                        remove_unsupported_columns(
+                            lower_case_keys(load_csv(p_person_file["path"])),
+                            table_name,
+                        )
+                    )
+                    for pp in pp_records:
+                        pp["is_primary"] = True
+                    records = pp_records + records
+
+                upsert_mutation = make_upsert_mutation(table_name)
+
+                for chunk in chunks(records, UPLOAD_BATCH_SIZE):
```

Comment:
Oh,  I'm seeing here that you are batching your graphql queries - this greatly nullifies my previous comment about latency. This is 👍